firestore/google/cloud/firestore_v1/query.py

```diff
@@ -968,3 +985,32 @@ def _query_response_to_snapshot(response_pb, collection, expected_prefix):
         update_time=response_pb.document.update_time,
     )
     return snapshot
+
+
+def _collection_group_query_response_to_snapshot(response_pb, collection):
+    """Parse a query response protobuf to a document snapshot.
```

Comment:
I am a little surprised that this is a different method from `_query_response_to_snapshot`.

Reply:
This was related to expectations built into this method. @tseaver made this to address that. It is possible we can combine them by refactoring that code.

Reply:
@schmidt-sebastian The implementation of `_query_response_to_snapshot` [mandates (via `_helpers.get_doc_id`)](https://github.com/googleapis/google-cloud-python/pull/7758/files#diff-56838955ee957b7f17f42e9e0d3925b0L959) that the result document is a direct child of the query's parent collection.  I'm unsure whether this restriction is "important", or just an artifact of prior development iterations.  If it is an artifact, then we could just  use the new implementation in `_collection_group_query_response_to_snapshot` for all queries.